src/netex/read.rs

```diff
@@ -15,9 +15,203 @@
 // <http://www.gnu.org/licenses/>.
 
 use model::Collections;
+use objects;
 use std::io::Read;
 use Result;
 
-pub fn read_netex_file<R: Read>(_collections: &mut Collections, _file: R) -> Result<()> {
-    Ok(())
+extern crate quick_xml;
+extern crate serde_json;
+use self::quick_xml::Reader;
+extern crate minidom;
+use self::minidom::Element;
+
+// type RoutePointId = String;
+// type StopPointId = String;
+// type RoutePointMapping = HashMap<RoutePointId, StopPointId>;
+// type RouteLineMap = HashMap<String, String>;
+
+#[derive(Default)]
+struct NetexContext {
+    namespace: String,
+    first_operator_id: String,
+    // network_id: String,
+    // routepoint_mapping: RoutePointMapping,
+    // route_line_map: RouteLineMap,
+    // route_mode_map: HashMap<String, String>,
+    // journeypattern_route_map: HashMap<String, String>,
+}
+
+pub struct NetexReader {
+    context: NetexContext,
+    pub collections: Collections,
+}
+impl NetexReader {
+    pub fn new() -> Self {
+        NetexReader {
+            context: NetexContext::default(),
+            collections: Collections::default(),
+        }
+    }
+
+    pub fn read_netex_file<R: Read>(&mut self, mut file: R) -> Result<()> {
+        let mut file_content = "".to_string();
+        file.read_to_string(&mut file_content)?;
+        let mut reader = Reader::from_str(&file_content);
+        let root = Element::from_reader(&mut reader)?;
```

Comment:
you can remove 
- `let mut reader = Reader::from_str(&file_content);`
- `extern crate quick_xml;`
- `use self::quick_xml::Reader;`
- `quick_xml` in Cargo.toml

and use `let root: Element = file_content.parse()?;`